skyportal/handlers/api/observingrun.py

```diff
@@ -0,0 +1,226 @@
+                joinedload(ObservingRun.assignments)
+                .joinedload(ClassicalAssignment.obj)
+                .joinedload(Obj.thumbnails)
+                .joinedload(Thumbnail.photometry),
+                joinedload(ObservingRun.assignments)
+                .joinedload(ClassicalAssignment.requester),
+                joinedload(ObservingRun.assignments)
+                .joinedload(ClassicalAssignment.obj)
+                .joinedload(Obj.comments),
+            ).filter(ObservingRun.id == run_id).first()
+
+            if run is None:
+                return self.error(f"Could not load observing run {run_id}",
+                                  data={"run_id": run_id})
+            # order the assignments by ra
+            run.assignments = sorted(
+                run.assignments, key=lambda a: a.obj.ra
+            )
+
+            # filter out the assignments of objects that are not visible to
+            # the user
+            run.assignments = list(filter(
+                lambda a: a.obj.is_owned_by(self.current_user),
+                run.assignments
+            ))
+
+            for assignment in run.assignments:
+                assignment.obj.comments = sorted(assignment.obj.comments,
+                                                 key=lambda c: c.modified,
+                                                 reverse=True)
+
+            data = ObservingRunGetWithAssignments.dump(run)
+            data['assignments'] = [a.to_dict() for a in data['assignments']]
+
+            # calculate when the targets rise and set
+            for d, a in zip(data['assignments'], run.assignments):
+                d['rise_time_utc'] = a.rise_time.isot
+                d['set_time_utc'] = a.set_time.isot
+
+            return self.success(data=data)
+        runs = ObservingRun.query.all()
+        data = ObservingRunGet.dump(runs, many=True)
+        out = sorted(data, key=lambda d: d['ephemeris']['sunrise_utc'])
+        return self.success(data=out)
+
+    @permissions(['Upload data'])
+    def put(self, run_id):
+        """
+        ---
+        description: Update observing run
+        parameters:
+          - in: path
+            name: run_id
+            required: true
+            schema:
+              type: integer
+        requestBody:
+          content:
+            application/json:
+              schema: ObservingRunPost
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+        run_id = int(run_id)
+        is_superadmin = 'System admin' in [a.id for a in self.current_user.acls]
+
+        orun = ObservingRun.query.get(run_id)
+
+        current_user_id = self.current_user.id if \
+            hasattr(self.current_user, 'username') else self.current_user.created_by_id
```

Comment:
```suggestion
        current_user_id = self.associated_user_object.id
```
I stand corrected - you can make multi-line suggestions!

Reply:
Fixed